inc/Debug/DebugHelper.php

```diff
@@ -83,6 +83,47 @@ public static function dbgDeprecatedProperty($class, $propertyName)
         );
     }
 
+    /**
+     * Trigger a custom deprecation event
+     *
+     * Usually dbgDeprecatedFunction or dbgDeprecatedProperty should be used instead.
+     * This method is only for those situation where more custom
+     *
+     * @param string $alternative
+     * @param string $deprecatedThing
+     * @param string $caller
+     * @param string $file
+     * @param int    $line
+     */
+    public static function dbgCustomDeprecationEvent(
+        $alternative,
+        $deprecatedThing,
+        $caller,
+        $file,
+        $line
```

Comment:
If it's OK to have this `dbgCustomDeprecationEvent()`, I would add `$callerOffset = 1` argument similar to `dbgDeprecatedFunction()` for more fine-grained control of the reported backtrace.

```suggestion
        $line,
        $callerOffset = 1
```

Looks like the current debug message is the most accurate one now.

```
::1: helper_plugin_move_handler::_finalize() is deprecated. It was called from dokuwiki\Parsing\Parser::parse in \dokuwiki\inc\Parsing\Parser.php:119 finalize() should be used instead!
```

Reply:
Done, I hope. Please have another look. I think `aaray_slice` should work here, but I'm a bit confused why we didn't use it in `dbgDeprecatedFunction()`?🤔